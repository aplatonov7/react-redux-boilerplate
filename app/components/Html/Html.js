import React, { PropTypes } from 'react'

const Html = ({ children, initialState = {}, scripts, styles, inlineStyles = '' }) => (
  <html lang="en">
    <head>
      <meta charSet="UTF-8" />
      <meta name="viewport" content="width=device-width, initial-scale=1.0" />
      <meta httpEquiv="X-UA-Compatible" content="ie=edge" />
      <title>My React-Redux app</title>
      {styles.map(s => <link rel="stylesheet" key={s} href={s} />)}
      {inlineStyles ? <style type="text/css" id="SSRStyles">{inlineStyles}</style> : null}
      <script
        dangerouslySetInnerHTML={{ __html: `
          window.__INITIAL_STATE__ = ${JSON.stringify(initialState)}
          window.__CLIENT__ = true
        ` }}
      />
    </head>
    <body>
      <div id="app">{children}</div>
      {scripts.map(s => <script key={s} src={s} />)}
      <script
        dangerouslySetInnerHTML={{ __html: 'document.getElementById("SSRStyles").remove()' }}
      />
    </body>
  </html>
)

Html.propTypes = {
  children: PropTypes.node,
  inlineStyles: PropTypes.string,
  initialState: PropTypes.object, // eslint-disable-line
  scripts: PropTypes.arrayOf(PropTypes.string).isRequired,
  styles: PropTypes.arrayOf(PropTypes.string).isRequired,
}

export default Html
