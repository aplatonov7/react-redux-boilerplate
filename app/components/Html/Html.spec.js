import React from 'react'
import renderer from 'react-test-renderer'
import Html from './'

describe('Html container', () => {
  const props = {
    scripts: ['test-script', 'test-script2'],
    styles: ['test-style', 'test-style2'],
    initialState: { test: 1 },
    inlineStyles: 'inline styles',
  }
  const item = renderer.create(<Html {...props}><div>child</div></Html>)

  it('should match snapshot', () => {
    expect(item.toJSON()).toMatchSnapshot()
  })
})
