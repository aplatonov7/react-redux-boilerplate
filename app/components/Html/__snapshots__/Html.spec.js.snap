exports[`Html container should match snapshot 1`] = `
<html
  lang="en">
  <head>
    <meta
      charSet="UTF-8" />
    <meta
      content="width=device-width, initial-scale=1.0"
      name="viewport" />
    <meta
      content="ie=edge"
      httpEquiv="X-UA-Compatible" />
    <title>
      My React-Redux app
    </title>
    <link
      href="test-style"
      rel="stylesheet" />
    <link
      href="test-style2"
      rel="stylesheet" />
    <style
      id="SSRStyles"
      type="text/css">
      inline styles
    </style>
    <script
      dangerouslySetInnerHTML={
        Object {
          "__html": "
                  window.__INITIAL_STATE__ = {\"test\":1}
                  window.__CLIENT__ = true
                ",
        }
      } />
  </head>
  <body>
    <div
      id="app">
      <div>
        child
      </div>
    </div>
    <script
      src="test-script" />
    <script
      src="test-script2" />
    <script
      dangerouslySetInnerHTML={
        Object {
          "__html": "document.getElementById(\"SSRStyles\").remove()",
        }
      } />
  </body>
</html>
`;
